### DISCUSSION TITLE
How to best implement Stork in Quarkus

### DISCUSSION THREAD

**Discussion by kjq** [smallrye/smallrye-stork#399] [Q&A]
We are moving to using Stork in Quarkus for our REST clients and have some questions about its usage and how to best implement it.  When running in development or testing we are using "static" resource and when deployed "kubernetes".

1. I am a little confused how to correctly register/discover resources across multiple services.  If I have Service A and Services B/C will use it - do I have to register A in both B/C services?

2. Does it make sense to have a centralized "discovery" service that manages all the services and then use a REST client to get the details in each of the consuming services which in turn would use that to instantiate the REST client?

**Comment by aureamunoz** [comment]
Hello!

Thank you for reaching out.

> 
>     1. I am a little confused how to correctly register/discover resources across multiple services.  If I have Service A and Services B/C will use it - do I have to register A in both B/C services?
> 

You should have a configuration for service B to discover service A and another config for service C to discover service A. If you are using Kubernetes service discovery, something like:
```
quarkus.stork.serviceB.service-discovery.type=kubernetes
quarkus.stork.serviceB.service-discovery.application=serviceA
```
and
```
quarkus.stork.serviceC.service-discovery.type=kubernetes
quarkus.stork.serviceC.service-discovery.type=serviceA
```

>     2. Does it make sense to have a centralized "discovery" service that manages all the services and then use a REST client to get the details in each of the consuming services which in turn would use that to instantiate the REST client?

Not sure to follow, maybe you need something like [Composite Service Discovery](http://smallrye.io/smallrye-stork/1.3.2/service-discovery/composite/)?
